lib/internal/process/per_thread.js

```diff
@@ -295,15 +298,16 @@ function buildAllowedFlags() {
   const nodeFlags = new SafeSet(ArrayPrototypeMap(allowedNodeEnvironmentFlags,
                                                   trimLeadingDashes));
 
-  class NodeEnvironmentFlagsSet extends SafeSet {
-    constructor(...args) {
-      super(...args);
+  class NodeEnvironmentFlagsSet extends Set {
+    #set;
+    constructor(iterable) {
+      super();
+      this.#set = new SafeSet(iterable);
+    }
 
-      // The super constructor consumes `add`, but
-      // disallow any future adds.
-      ObjectDefineProperty(this, 'add', {
-        value: () => this
-      });
+    add() {
+      // No-op, `Set` API compatible
```

Comment:
It might seem silly but we need to at least consider whether there needs to be a deprecation cycle on this before it's made non-op.

Reply:
Note that `process.allowedNodeEnvironmentFlags.add` is already no-op on all Node.js release lines.
We may or may not consider it a breaking change to make `Set.prototype.add.call(process.allowedNodeEnvironmentFlags)` no-op – IMHO we should not as this never was a documented thing, but no strong feelings.